### ISSUE TITLE
Unable to open observatory when run from XCode

### ISSUE THREAD

**Issue by ristiisa** [flutter/devtools#3902]
When flutter application is run on a physical device from terminal or from vs code the observatory URL that is printed out can be opened. But when the flutter application is launched from XCode, the observatory URL that is printed to debug output refuses to connect. When IOS simulator is used then everything works as expected.

```
[✓] Flutter (Channel stable, 2.10.3, on macOS 11.6.2 20G314 darwin-x64, locale en-EE)
    • Flutter version 2.10.3
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (3 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK
    • Platform android-31, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.64.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[!] Proxy Configuration
    • HTTP_PROXY is set
    ! NO_PROXY is not set

[✓] Connected device (5 available)
    • iPhone (mobile)            • 00008030-000269E13EBA802E            • ios            • iOS 15.3.1 19D52
    • iPhone SE (2nd generation) (mobile) • 44C45F6B-C034-491C-8607-8F9D05131829 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • iPhone 13 mini (mobile)             • A842F941-DB2C-48B4-B053-6EEF25329C8D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)                        • chrome                               • web-javascript • Google Chrome 99.0.4844.74

[✓] HTTP Host Availability
    • All required HTTP hosts are available
```

**Comment by kenzieschmoll** [comment]
There are two URLs printed to command line when `flutter run` is executed. Are you talking about the Observatory URL or the DevTools URL? We recommend using DevTools for debugging and profiling your flutter app.
```
$ flutter run 
Launching lib/main.dart on iPhone 13 in debug mode...
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         7.0s
Xcode build done.                                           19.5s
Syncing files to device iPhone 13...                               165ms

...

An Observatory debugger and profiler on iPhone 13 is available at: http://127.0.0.1:58847/AxWIbuQy-oA=/
The Flutter DevTools debugger and profiler on iPhone 13 is available at:
http://127.0.0.1:9103?uri=http://127.0.0.1:58847/AxWIbuQy-oA=/
```

Additionally, are you copying and pasting the URL into a browser or clicking a link and letting XCode handle the browser launch? If you are still seeing this issue when copying the url and pasting into a browser, then perhaps there is an issue with how XCode is interacting with Flutter tools. @christopherfujino - do we have guidance for users that are attempting to launch their flutter app directly from XCode? Do we expect flutter tools to behave differently in this situation?

**Comment by kenzieschmoll** [comment]
Without further response, we cannot make progress on resolving this issue. Please re-open if you are still seeing this problem.
